### PR TITLE
[RFC][ADMIN-PANEL-2] Export frontend as package

### PR DESCRIPTION
**Pull request by ogp-weeloong**
# Problem
We want to re-use front-end components in our admin panel

# Solution
We will convert our frontend to a package so that we can export our frontend end components.

This uses vite lib mode to build the package; we bundle it up because we still want to deploy and serve this over the network.

Note that this PR has a lot of files because it changes the frontend to use `@/` prefix for root path alias; we were previously using an empty prefix. This is because I need to fix the import paths in the generated `d.ts` files, but I couldn't get `tsc-alias` to work with a empty prefix. <.<

# Tests
- Check that I can install frontend as a package (via `npm pack` then `file` dependency in another repo)
- Regression test: check that frontend still works (can create pipe, query pipe, etc).
- Regression test: check that builds still work (deploy to UAT and verify it works)

### CODE REVIEW

**Comment by ogp-weeloong** [comment]
* **#677** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/677?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#672** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/672?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#673** <a href="https://app.graphite.dev/github/pr/opengovsg/plumber/673?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop-v2`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ogp-weeloong and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/src/exports/dependencies/index.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import * as apollo from './apollo'
+import * as chakra from './chakra'
+import * as graphql from './graphql'
+import * as ogpDesignSystem from './ogp-design-system'
+import * as reactRouter from './react-router'
+
```
Comment (pregnantboy) [inline packages/frontend/src/exports/dependencies/index.ts]: should these be peer deps instead?

File: packages/frontend/vite.config.lib.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        ),
+        'exports/dependencies/react-router': resolve(
+          __dirname,
+          'src/exports/dependencies/react-router.ts',
+        ),
+        'exports/dependencies/ogp-design-system': resolve(
+          __dirname,
+          'src/exports/dependencies/ogp-design-system.ts',
+        ),
+      },
```
Comment (pregnantboy) [inline packages/frontend/vite.config.lib.ts]: do we need the extra entrypoints if they're all exported by index?

**Comment by pregnantboy** [comment]
does it not work by leaving 
```
"baseUrl": "./src",
```
in tsconfig without adding prefixes?

**Review by pregnantboy — APPROVED** [review]

**Comment by ogp-weeloong** [comment]
### Merge activity

* **Jul 31, 9:04 AM EDT**: @ogp-weeloong started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/672).
* **Jul 31, 9:07 AM EDT**: [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/672) couldn't merge this PR because it had conflicts with the trunk branch.
* **Jul 31, 9:23 AM EDT**: @ogp-weeloong started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/672).
* **Jul 31, 9:23 AM EDT**: @ogp-weeloong merged this pull request with [Graphite](https://app.graphite.dev/github/pr/opengovsg/plumber/672).

**Review by m0nggh — COMMENTED** [review]
damn sick thks for fixing all the import paths wow

**Pull request merged**